compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -17,21 +17,47 @@ namespace mlir::iree_compiler {
 /// approximation.
 static llvm::cl::opt<bool> clNativeMathPrecision(
     "iree-codegen-gpu-native-math-precision",
-    llvm::cl::desc(
-        "Skip polynomial lowering for math op natively available on GPU"),
+    llvm::cl::desc("Deprecated! This flag had buggy/unintentional semantics. "
+                   "Its original description said: \"Skip polynomial lowering "
+                   "for math op natively available on GPU.\""),
     llvm::cl::init(false));
 
 #define GEN_PASS_DEF_POLYNOMIALAPPROXIMATIONPASS
 #include "iree/compiler/Codegen/Common/Passes.h.inc"
 
 namespace {
 
-static void populateErfPattern(RewritePatternSet &patterns) {
-  if (clNativeMathPrecision) {
-    patterns.add<math::ErfPolynomialApproximation>(patterns.getContext());
-  } else {
+static void populateMathFunctionsExpandPatterns(
+    RewritePatternSet &patterns,
+    const std::function<bool(StringRef)> &predicate) {
+  if (predicate(math::TanOp::getOperationName())) {
+    populateExpandTanPattern(patterns);
+  }
+  if (predicate(math::SinhOp::getOperationName())) {
+    populateExpandSinhPattern(patterns);
+  }
+  if (predicate(math::CoshOp::getOperationName())) {
+    populateExpandCoshPattern(patterns);
+  }
+  if (predicate(math::AsinhOp::getOperationName())) {
+    populateExpandAsinhPattern(patterns);
+  }
+  if (predicate(math::AcoshOp::getOperationName())) {
+    populateExpandAcoshPattern(patterns);
+  }
+  if (predicate(math::AtanhOp::getOperationName())) {
+    populateExpandAtanhPattern(patterns);
+  }
+  if (predicate(math::PowFOp::getOperationName())) {
+    populateExpandPowFPattern(patterns);
+  }
+  if (predicate(math::FPowIOp::getOperationName())) {
+    populateExpandFPowIPattern(patterns);
+  }
```

Comment:
Suggestion: these are redundant code. try to refactor into a function and a map like here: https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp#L50-L59